src/v2/models/AppState.js

```diff
@@ -241,33 +243,47 @@ export default Model.extend({
     this.trigger('cache:clear');
   },
 
-  setIonResponse(transformedResponse) {
+  async setIonResponse(transformedResponse, hooks) {
     if (!this.shouldReRenderView(transformedResponse)) {
       return;
     }
 
-    // `currentFormName` is default to first form of remediations or nothing.
-    let currentFormName = null;
-    if (!_.isEmpty(transformedResponse.remediations)) {
-      currentFormName = transformedResponse.remediations[0].name;
-    } else {
-      Logger.error('Panic!!');
-      Logger.error('\tNo remediation found.');
-      Logger.error('\tHere is the entire response');
-      Logger.error(JSON.stringify(transformedResponse, null, 2));
-    }
+    const currentFormName = this.getFormName(transformedResponse);
 
     this.clearAppStateCache();
 
     // set new app state properties
     this.set(transformedResponse);
 
+    const hook = hooks?.getHook(currentFormName); // may be undefined
+    await executeHooksBefore(hook);
```

Comment:
Instead of widget always execute the hooks.. should we relay this control of execution to the individual flows which expect hooks to be executed.. So when these flows renders screens, they can instruct widget to execute the flow `preRender` or `postRender` ?

Reply:
My expectation is that a common use of this feature will be for adding tracking or logging, so it makes sense to have it active on all views.